app/tests/test_date_validator.py

```diff
@@ -0,0 +1,363 @@
+                {"date_from": "01/10/2023", "date_to": current_date()},
+            ),
+            (
+                "date_from_day=&date_from_month=02&date_from_year=2020",
+                {"date_from": "01/02/2020", "date_to": current_date()},
+            ),
+            (
+                "date_from_day=&date_from_month=&date_from_year=2023",
+                {"date_from": "01/01/2023", "date_to": "31/12/2023"},
+            ),
+            (
+                "date_from_day=&date_from_month=&date_from_year=2024",
+                {"date_from": "01/01/2024", "date_to": current_date()},
+            ),
+            (
+                "date_from_day=&date_from_month=03&date_from_year=2024",
+                {"date_from": "01/03/2024", "date_to": current_date()},
+            ),
+            (
+                "date_from_day=01&date_from_month=10&date_from_year=2023",
+                {"date_from": "01/10/2023", "date_to": current_date()},
+            ),
+            (
+                "date_to_day=&date_to_month=03&date_to_year=2023",
+                {
+                    "date_from": default_start_date().strftime(
+                        python_date_format
+                    ),
+                    "date_to": "31/03/2023",
+                },
+            ),
+            (
+                "date_to_day=&date_to_month=02&date_to_year=2024",
+                {
+                    "date_from": default_start_date().strftime(
+                        python_date_format
+                    ),
+                    "date_to": "29/02/2024",
+                },
+            ),
+            (
+                "date_to_day=&date_to_month=&date_to_year=2023",
+                {
+                    "date_from": default_start_date().strftime(
+                        python_date_format
+                    ),
+                    "date_to": "31/12/2023",
+                },
+            ),
+            (
+                "date_to_day=31&date_to_month=03&date_to_year=2023",
+                {
+                    "date_from": default_start_date().strftime(
+                        python_date_format
+                    ),
+                    "date_to": "31/03/2023",
+                },
+            ),
+            (
+                "date_from_day=&date_from_month=&date_from_year=2023&"
+                "date_to_day=&date_to_month=&date_to_year=2023",
+                {"date_from": "01/01/2023", "date_to": "31/12/2023"},
+            ),
+            (
+                "date_from_day=01&date_from_month=08&date_from_year=2023&"
+                "date_to_day=31&date_to_month=10&date_to_year=2023",
+                {"date_from": "01/08/2023", "date_to": "31/10/2023"},
+            ),
+            (
+                "date_from_day=&date_from_month=&date_from_year=&"
+                "date_to_day=&date_to_month=&date_to_year=",
+                {"date_from": "##/##/####", "date_to": "##/##/####"},
```

Comment:
why are we returning hashes?

Reply:
to show the correct values on date filter inputs are we are reading [0:2], [3:5] , [6:10] values in date-filters.html